### PR TITLE
FLA-1234 improved logging of diligen calls

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

https://justice.gov.bc.ca/jira/browse/FLA-1234

Enhanced logging of some diligen calls:
- If document validation fails, the logger will output as a warning, ie:
`document 0 failed validation: {"type":"DOCUMENT_TYPE","expected":"Response to Civil Claim","actual":"THIS IS NOT VALID"}`
This should capture the criteria to log document type mismatches, or other type of validation failures.

- Added a timer around a diligen call to track how long a submission takes, ie.
`document submission to diligen start, filename: test-document.pdf`
`document submission to diligen finish, filename: test-document.pdf, size: 58101 bytes, time: 1287 ms`

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/aireviewerapi/document/DocumentsApiDelegateImpl.java [hunk 1]
```diff
@@ -98,7 +100,11 @@ public DocumentsApiDelegateImpl(
         //Due to a bug in mapstruct logic has been performed inline
         ExtractRequest extractRequest = extractRequestMapper.toExtractRequest(extractRequestMapper.toExtract(xTransactionId, (xUseWebhook == null || xUseWebhook)), xDocumentType, file, receivedTimeMillis);
 
+        logger.info("document submission to diligen start, filename: " + file.getName());
+        stopWatch.start();
         BigDecimal response = diligenService.postDocument(xDocumentType, file, documentTypeConfiguration.getProjectId());
+        stopWatch.stop();
+        logger.info("document submission to diligen finish, filename: " + file.getName() + ", size: " + file.getSize() + " bytes, time: " + stopWatch.getLastTaskTimeMillis() + " ms");
```
Comment (TayGov) [inline src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/aireviewerapi/document/DocumentsApiDelegateImpl.java:107]: Nice this covers it. I think we want to move it to handleDocumentEvent. There is a total processing time in the EventRequest Object. We may want to use transaction id as well. My only concern with filename it we don't control it. We may want to ask Jaime if there is ever a persons name in it.

Comment (KDNeufeld) [reply src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/aireviewerapi/document/DocumentsApiDelegateImpl.java:107]: Sounds good.  On it. I was wondering about the filename myself ... don't know if it's considered PI

File: src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/aireviewerapi/document/DocumentsApiDelegateImpl.java [hunk 2]
```diff
@@ -98,7 +100,11 @@ public DocumentsApiDelegateImpl(
         //Due to a bug in mapstruct logic has been performed inline
         ExtractRequest extractRequest = extractRequestMapper.toExtractRequest(extractRequestMapper.toExtract(xTransactionId, (xUseWebhook == null || xUseWebhook)), xDocumentType, file, receivedTimeMillis);
 
+        logger.info("document submission to diligen start, filename: " + file.getName());
```
Comment (BrianMaki) [inline src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/aireviewerapi/document/DocumentsApiDelegateImpl.java:103]: The logger call can be cleaned up a little further by using args implementation like logger.info("document submission to diligen start, filename: {}", file.getName()

File: src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/aireviewerapi/document/DocumentsApiDelegateImpl.java (hunk 1 above)
Comment (BrianMaki) [inline src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/aireviewerapi/document/DocumentsApiDelegateImpl.java:107]: The logger call can be cleaned up a little further by using args implementation like logger.info("document submission to diligen finish, filename: {}, size: {} bytes, time: {} ms)

Comment (KDNeufeld) [reply src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/aireviewerapi/document/DocumentsApiDelegateImpl.java]: Moved to handleDocumentEvent and simply enhanced the logging there to also include transactionID.

File: src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/aireviewerapi/document/DocumentsApiDelegateImpl.java (hunk 2 above)
Comment (KDNeufeld) [reply src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/aireviewerapi/document/DocumentsApiDelegateImpl.java]: Good.  Done.

**Review by TayGov — APPROVED** [review]

**Pull request merged**